packages/redux/src/analytics/middlewares/__tests__/bag.test.ts

```diff
@@ -211,59 +219,68 @@ describe('bagMiddleware()', () => {
   });
 
   describe('Remove item from bag', () => {
-    let bagItem;
+    let bagItem: BagItemEntity;
 
     beforeEach(() => {
       const mockState = getMockState();
 
       // Test promotions with a value greater than 0
-      bagItem = merge({}, mockState.entities.bagItems[mockBagItemId]);
+      bagItem = merge(
+        {},
+        bagMockData.mockState?.entities?.bagItems?.[bagMockData.mockBagItemId],
+      ) as BagItemEntity;
       bagItem.quantity = 5;
       bagItem.promotionDetail.totalDiscountValue = 100;
-      mockState.entities.bagItems[mockBagItemId] = bagItem;
 
-      bagItem = mockState.entities.bagItems[mockBagItemId];
+      // @ts-ignore this entity aren't nullable
+      mockState.entities.bagItems[bagMockData.mockBagItemId] = bagItem;
+
+      // @ts-ignore this entity aren't nullable
```

Comment:
This comment is not necessary if you access the prop just like it is being done elsewhere, like this:

```js
bagItem = mockState.entities?.bagItems?.[
        bagMockData.mockBagItemId
      ] as BagItemEntity;
```